### PR TITLE
Consume the internalized System.Net.Quic sources

### PR DESCRIPTION
**Pull request by Tratcher**
This mirrors https://github.com/dotnet/runtime/pull/32549.
@scalablecory I've also pulled in your most recent changes to the shared sources. See the second commit. Fixes #18943

@jkotalik Please help me run a sanity test with MSQuic.

### CODE REVIEW

File: src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+            // platform-specific errors in subsequent callbacks. For now, check for the minimum build we've tested it on.
+
+            // TODO:
+            // - Hopefully, MsQuicOpen will perform this check for us and give us a consistent error code.
+            // - Otherwise, dial this in to reflect actual minimum requirements and add some sort of platform
+            //   error code mapping when creating exceptions.
+
+            OperatingSystem ver = Environment.OSVersion;
+
+            if (ver.Platform == PlatformID.Win32NT && ver.Version < new Version(10, 0, 19041, 0))
```
Comment (jkotalik) [inline src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs]: This check fails in our repo.

**Review by jkotalik — APPROVED** [review]
Pending the OS check, lgtm

File: src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs (hunk 1 above)
Comment (Tratcher) [reply src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs]: @scalablecory this is something we'll need to revisit. Environment.OSVersion is extremely unreliable and shouldn't be used in production code. I'll file an issue.

Comment (Tratcher) [reply src/Shared/runtime/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs]: https://github.com/dotnet/runtime/issues/32575

**Comment by wtgodbe** [comment]
@Tratcher is this ready to go? If so please merge

**Comment by Tratcher** [comment]
@wtgodbe this is now ready to merge but the branch is restricted. Please merge.

**Pull request merged**